.github/unittest.sh

```diff
@@ -7,6 +7,30 @@ CONDA_PATH=$(which conda)
 eval "$(${CONDA_PATH} shell.bash hook)"
 conda config --set channel_priority strict
 
+# Setup the OS_TYPE environment variable that should be used for conditions involving the OS below.
+case $(uname) in
+  Linux)
+    OS_TYPE=linux
+    ;;
+  Darwin)
+    OS_TYPE=macos
+    ;;
+  *)
+    echo "Unknown OS type:" $(uname)
+    exit 1
+    ;;
+esac
+
+# The x86 macOS runners, e.g. the GitHub Actions native "macos-12" runner, has some JPEG libraries installed by default
```

Comment:
Let's wrap this with the GHA `::group::` so it's clear what's happing in the logs